apis/goharbor.io/v1beta1/chartmuseum_types.go

```diff
@@ -0,0 +1,361 @@
+// ChartMuseumList contains a list of ChartMuseum.
+type ChartMuseumList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []ChartMuseum `json:"items"`
+}
+
+// ChartMuseumSpec defines the desired state of ChartMuseum.
+type ChartMuseumSpec struct {
+	harbormetav1.ComponentSpec `json:",inline"`
+
+	CertificateInjection `json:",inline"`
+
+	// +kubebuilder:validation:Optional
+	Log ChartMuseumLogSpec `json:"log,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	Authentication ChartMuseumAuthSpec `json:"authentication,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	Server ChartMuseumServerSpec `json:"server,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// Disable some features
+	Disable ChartMuseumDisableSpec `json:"disable,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// Cache stores
+	Cache ChartMuseumCacheSpec `json:"cache,omitempty"`
+
+	// +kubebuilder:validation:Required
+	Chart ChartMuseumChartSpec `json:"chart"`
+}
+
+type ChartMuseumServerSpec struct {
+	// +kubebuilder:validation:Optional
+	TLS *harbormetav1.ComponentsTLSSpec `json:"tls,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Type="string"
+	// +kubebuilder:validation:Pattern="([0-9]+h)?([0-9]+m)?([0-9]+s)?([0-9]+ms)?([0-9]+us)?([0-9]+µs)?([0-9]+ns)?"
+	// Socket timeout
+	ReadTimeout *metav1.Duration `json:"readTimeout,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Type="string"
+	// +kubebuilder:validation:Pattern="([0-9]+h)?([0-9]+m)?([0-9]+s)?([0-9]+ms)?([0-9]+us)?([0-9]+µs)?([0-9]+ns)?"
+	// Socket timeout
+	WriteTimeout *metav1.Duration `json:"writeTimeout,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:default=20971520
+	// Max size of post body (in bytes)
+	MaxUploadSize *int64 `json:"maxUploadSize,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// Value to set in the Access-Control-Allow-Origin HTTP header
+	CORSAllowOrigin string `json:"corsAllowOrigin,omitempty"`
+}
+
+type ChartMuseumChartSpec struct {
+	// +kubebuilder:validation:Optional
+	// Form fields which will be queried
+	PostFormFieldName ChartMuseumPostFormFieldNameSpec `json:"postFormFieldName,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Pattern="https?://.*"
+	// The absolute url for .tgz files in index.yaml
+	URL string `json:"url,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default=true
+	// Allow chart versions to be re-uploaded without ?force querystring
+	AllowOverwrite *bool `json:"allowOverwrite,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default=false
+	// Enforce the chart museum server only accepts the valid chart version as Helm does
+	SemanticVersioning2Only bool `json:"onlySemver2"`
+
+	// +kubebuilder:validation:Required
+	Storage ChartMuseumChartStorageSpec `json:"storage"`
```

Comment:
It seems to be different from what was discussed at the weekly meeting
It should be an alias for v1alpha3
```
type ChartMuseumChartStorageSpec harborv1alpha3.
```

Reply:
At the weekly meeting, We reached a compromise, the types are completely copied from v1alpha3, and the conversion using json copy.